pkg/operator/mon/mon_test.go

```diff
@@ -81,6 +88,139 @@ func TestStartMonPods(t *testing.T) {
 	validateStart(t, c)
 }
 
+func TestOperatorRestart(t *testing.T) {
+	clientset := test.New(3)
+	namespace := "ns"
+	configDir, _ := ioutil.TempDir("", "")
+	defer os.RemoveAll(configDir)
+	executor := &exectest.MockExecutor{
+		MockExecuteCommandWithOutput: func(debug bool, actionName string, command string, args ...string) (string, error) {
+			if strings.Contains(command, "ceph-authtool") {
+				cephtest.CreateClusterInfo(nil, path.Join(configDir, namespace), nil)
+			}
+			return "", nil
+		},
+	}
+	context := &clusterd.Context{
+		Clientset: clientset,
+		Executor:  executor,
+		ConfigDir: configDir,
+	}
+	c := &Cluster{
+		context:             context,
+		Namespace:           namespace,
+		Version:             "myversion",
+		Size:                3,
+		maxMonID:            -1,
+		waitForStart:        false,
+		monPodRetryInterval: 10 * time.Millisecond,
+		monPodTimeout:       1 * time.Second,
+		monTimeoutList:      map[string]time.Time{},
+		mapping: &mapping{
+			Node: map[string]*nodeInfo{},
+			Port: map[string]int32{},
+		},
+	}
+
+	// start a basic cluster
+	info, err := c.Start()
+	assert.Nil(t, err)
+	assert.NotNil(t, info)
+
+	validateStart(t, c)
+
+	c = &Cluster{
+		context:             context,
+		Namespace:           namespace,
+		Version:             "myversion",
+		Size:                3,
+		maxMonID:            -1,
+		waitForStart:        false,
+		monPodRetryInterval: 10 * time.Millisecond,
+		monPodTimeout:       1 * time.Second,
+		monTimeoutList:      map[string]time.Time{},
+		mapping: &mapping{
+			Node: map[string]*nodeInfo{},
+			Port: map[string]int32{},
+		},
+	}
+
+	// starting again should be a no-op, but still results in an error
+	info, err = c.Start()
+	assert.Nil(t, err)
+	assert.NotNil(t, info)
+
+	validateStart(t, c)
+}
+
+// safety check that if hostNetwork is used no changes occur on an operator restart
+func TestOperatorRestartHostNetwork(t *testing.T) {
```

Comment:
is this test exactly the same except for the HostNetwork being true? would be nice to factor to a helper method

Reply:
I replaced 108 lines with 27 lines. It should be better now :)